autogen/agentchat/contrib/qdrant_retrieve_user_proxy_agent.py

```diff
@@ -0,0 +1,225 @@
+                will be used. If you want to use other vector db, extend this class and override the `retrieve_docs` function.
+            - docs_path (Optional, str): the path to the docs directory. It can also be the path to a single file,
+                or the url to a single file. Default is None, which works only if the collection is already created.
+            - collection_name (Optional, str): the name of the collection.
+                If key not provided, a default name `autogen-docs` will be used.
+            - model (Optional, str): the model to use for the retrieve chat.
+                If key not provided, a default model `gpt-4` will be used.
+            - chunk_token_size (Optional, int): the chunk token size for the retrieve chat.
+                If key not provided, a default size `max_tokens * 0.4` will be used.
+            - context_max_tokens (Optional, int): the context max token size for the retrieve chat.
+                If key not provided, a default size `max_tokens * 0.8` will be used.
+            - chunk_mode (Optional, str): the chunk mode for the retrieve chat. Possible values are
+                "multi_lines" and "one_line". If key not provided, a default mode `multi_lines` will be used.
+            - must_break_at_empty_line (Optional, bool): chunk will only break at empty line if True. Default is True.
+                If chunk_mode is "one_line", this parameter will be ignored.
+            - embedding_model (Optional, str): the embedding model to use for the retrieve chat.
+                If key not provided, a default model `bge-small-en-v1.5` will be used. All available models
+                can be found at `https://qdrant.github.io/fastembed/examples/Supported_Models/`.
+            - customized_prompt (Optional, str): the customized prompt for the retrieve chat. Default is None.
+            - customized_answer_prefix (Optional, str): the customized answer prefix for the retrieve chat. Default is "".
+                If not "" and the customized_answer_prefix is not in the answer, `Update Context` will be triggered.
+            - update_context (Optional, bool): if False, will not apply `Update Context` for interactive retrieval. Default is True.
+            - custom_token_count_function(Optional, Callable): a custom function to count the number of tokens in a string.
+                The function should take a string as input and return three integers (token_count, tokens_per_message, tokens_per_name).
+                Default is None, tiktoken will be used and may not be accurate for non-OpenAI models.
+            - custom_text_split_function(Optional, Callable): a custom function to split a string into a list of strings.
+                Default is None, will use the default function in `autogen.retrieve_utils.split_text_to_chunks`.
+        **kwargs (dict): other kwargs in [UserProxyAgent](../user_proxy_agent#__init__).
+
+    """
```

Comment:
Should this go to __init__?

Reply:
Right. Will do.

Reply:
https://github.com/microsoft/autogen/pull/303/commits/3ea410b98de09fdd07c76feb32854f5d265527ae